pkg/assets/crd.go

```diff
@@ -44,57 +42,82 @@ var (
 	}
 )
 
-func getCRD(clientset *apiext_clientset.Clientset, obj apiruntime.Object) error {
+func isEstablished(clientset *apiext_clientset.Clientset, obj apiruntime.Object) (bool, error) {
 	gv := obj.GetObjectKind().GroupVersionKind().GroupVersion()
 	switch gv.String() {
 	case "apiextensions.k8s.io/v1":
 		v1Obj := obj.(*apiextv1.CustomResourceDefinition)
-		_, err := clientset.ApiextensionsV1().CustomResourceDefinitions().Get(context.TODO(), v1Obj.Name, metav1.GetOptions{})
+		crd, err := clientset.ApiextensionsV1().CustomResourceDefinitions().Get(context.TODO(), v1Obj.Name, metav1.GetOptions{})
 		if err != nil {
 			lastErr := fmt.Errorf("error getting CustomResourceDefinition %s: %v", v1Obj.Name, err)
 			logrus.Infof("getting openshift CRD status %v", lastErr)
-			return lastErr
+			if apierrors.IsNotFound(err) {
+				return false, nil
+			}
+			return false, lastErr
+		}
+		for _, condition := range crd.Status.Conditions {
+			if condition.Type == apiextv1.Established && condition.Status == apiextv1.ConditionTrue {
+				return true, nil
+			}
 		}
 	case "apiextensions.k8s.io/v1beta1":
 		v1beta1Obj := obj.(*apiextv1beta1.CustomResourceDefinition)
-		_, err := clientset.ApiextensionsV1beta1().CustomResourceDefinitions().Get(context.TODO(), v1beta1Obj.Name, metav1.GetOptions{})
+		crd, err := clientset.ApiextensionsV1beta1().CustomResourceDefinitions().Get(context.TODO(), v1beta1Obj.Name, metav1.GetOptions{})
 		if err != nil {
 			lastErr := fmt.Errorf("error getting CustomResourceDefinition %s: %v", v1beta1Obj.Name, err)
 			logrus.Infof("getting openshift CRD status %v", lastErr)
-			return lastErr
+			if apierrors.IsNotFound(err) {
```

Comment:
maybe some refactoring here to check err and `crd.Status.Conditions` in one block

Reply:
Yeah, it's not super readable. Will clean up.